xarray/core/types.py

```diff
@@ -64,7 +64,8 @@
 VarCompatible = Union["Variable", "ScalarOrArray"]
 GroupByIncompatible = Union["Variable", "GroupBy"]
 
-Dims = Union[str, Iterable[Hashable], None]
+Dims = Union[str, Iterable[Hashable], "ellipsis", None]
+OrderedDims = Union[str, Sequence[Union[Hashable, "ellipsis"]], "ellipsis", None]
```

Comment:
I realize the work has been done already, but would it be simpler to just always have `OrderedDims`, and never allow just an `Iterable`? While there are a couple of methods (e.g. `drop`) that can take something unordered, it might be fine to only take a Sequence.

Reply:
Personally I like using sets for dimensions. Basically all reductions can work with unordered dimensions.

Reply:
Yes good point!